cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1245,13 +1195,31 @@ void writer::impl::encode_pages(hostdevice_2dvector<gpu::EncColumnChunk>& chunks
   switch (compression_) {
     case parquet::Compression::SNAPPY:
       if (nvcomp_integration::is_stable_enabled()) {
-        snappy_compress(comp_in, comp_out, comp_stats, max_page_uncomp_data_size, stream);
+        nvcomp::batched_compress(nvcomp::compression_type::SNAPPY,
+                                 comp_in,
+                                 comp_out,
+                                 comp_stats,
+                                 max_page_uncomp_data_size,
+                                 stream);
       } else {
         gpu_snap(comp_in, comp_out, comp_stats, stream);
       }
       break;
-    default: break;
+    case parquet::Compression::ZSTD:
+      if (nvcomp_integration::is_all_enabled()) {
+        nvcomp::batched_compress(nvcomp::compression_type::ZSTD,
+                                 comp_in,
+                                 comp_out,
+                                 comp_stats,
+                                 max_page_uncomp_data_size,
+                                 stream);
+      }
+      break;
+    case parquet::Compression::UNCOMPRESSED: break;
+    default: CUDF_FAIL("invalid compression type");
   }
+  compress_check(comp_stats, stream);
```

Comment:
Does it make sense to call this for parquet::Compression::UNCOMPRESSED?

Reply:
It's safe to call because `comp_stats` will be empty in that case.